node/src/p2p/callbacks/p2p_callbacks_reducer.rs

```diff
@@ -0,0 +1,384 @@
+                                        right.clone(),
+                                    ),
+                                },
+                            );
+                        }
+                        MinaLedgerSyncLedgerAnswerStableV2::ContentsAre(accounts) => {
+                            dispatcher.push(
+                                TransitionFrontierSyncLedgerSnarkedAction::PeerQueryAddressSuccess {
+                                    peer_id,
+                                    rpc_id: id,
+                                    response: PeerLedgerQueryResponse::ChildAccounts(
+                                        accounts.iter().cloned().collect(),
+                                    ),
+                                },
+                            );
+                        }
+                        MinaLedgerSyncLedgerAnswerStableV2::NumAccounts(count, contents_hash) => {
+                            dispatcher.push(
+                                TransitionFrontierSyncLedgerSnarkedAction::PeerQueryNumAccountsSuccess {
+                                    peer_id,
+                                    rpc_id: id,
+                                    response: PeerLedgerQueryResponse::NumAccounts(
+                                        count.as_u64(), contents_hash.clone()
+                                    ),
+                                },
+                            );
+                        }
+                    },
+                    Some(P2pRpcResponse::StagedLedgerAuxAndPendingCoinbasesAtBlock(parts)) => {
+                        dispatcher.push(
+                            TransitionFrontierSyncLedgerStagedAction::PartsPeerFetchSuccess {
+                                peer_id,
+                                rpc_id: id,
+                                parts: parts.clone(),
+                            },
+                        );
+                    }
+                    Some(P2pRpcResponse::Block(block)) => {
+                        let Ok(block) = BlockWithHash::try_new(block.clone()) else {
+                            openmina_core::error!(meta.time(); "P2pRpcResponse::Block: invalid block");
+                            return;
+                        };
+                        dispatcher.push(TransitionFrontierSyncAction::BlocksPeerQuerySuccess {
+                            peer_id,
+                            rpc_id: id,
+                            response: block,
+                        });
+                    }
+                    Some(P2pRpcResponse::Snark(snark)) => {
+                        dispatcher.push(SnarkPoolCandidateAction::WorkReceived {
+                            peer_id,
+                            work: snark.clone(),
+                        });
+                    }
+                    Some(P2pRpcResponse::InitialPeers(_)) => {}
+                }
+                dispatcher.push(TransitionFrontierSyncLedgerSnarkedAction::PeersQuery);
```

Comment:
Not an issue with the PR, the old version already had this bug. But in the cases in which the flow is short-circuited by a `return` because of an error, these dispatches will never happen.

Reply:
Cannot come up with an obvious solution right now, but lets at least add a `FIXME` comment here mentioning that these will not be dispatched in the case of errors.

Reply:
Well, actually, there is an obvious solution. The body of these match branch is too long, the handling could be moved to an outer helper function/method, then the `return` will not be an issue.